// [[Reason: Declarations types can't be named so can't be fixed]] ////

//// [tests/cases/conformance/types/thisType/declarationFiles.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,13 +1,58 @@
 
+
+//// [declarationFiles.d.ts]
+declare class C1 {
+    x: this;
+    f(x: this): this;
+    constructor(x: this);
+}
+declare class C2 {
+    [x: string]: this;
+}
+interface Foo<T> {
+    x: T;
+    y: this;
+}
+declare class C3 {
+    a: this[];
+    b: [this, this];
+    c: this | Date;
+    d: this & Date;
+    e: (((this)));
+    f: (x: this) => this;
+    g: new (x: this) => this;
+    h: Foo<this>;
+    i: Foo<this | (() => this)>;
+    j: (x: any) => x is this;
+}
+declare const x3_a: typeof globalThis;
+declare const x1_a: typeof globalThis;
+declare class C4 {
+    x1: {
+        a: typeof x1_a;
+    };
+    x2: this[];
+    x3: readonly [{
+        readonly a: typeof x3_a;
+    }];
+    x4: () => this;
+    f1(): invalid;
+    f2(): this[];
+    f3(): invalid;
+    f4(): () => this;
+}
+//# sourceMappingURL=declarationFiles.d.ts.map
 /// [Errors] ////
 
 declarationFiles.ts(4,20): error TS2526: A 'this' type is available only in a non-static member of a class or interface.
 declarationFiles.ts(36,5): error TS2527: The inferred type of 'f1' references an inaccessible 'this' type. A type annotation is necessary.
+declarationFiles.ts(36,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
 declarationFiles.ts(42,5): error TS2527: The inferred type of 'f3' references an inaccessible 'this' type. A type annotation is necessary.
+declarationFiles.ts(42,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
 
 
-==== declarationFiles.ts (3 errors) ====
+==== declarationFiles.ts (5 errors) ====
     class C1 {
         x: this;
         f(x: this): this { return undefined; }
         constructor(x: this) { }
@@ -46,16 +91,22 @@
         x4 = (): this => this;
         f1() {
         ~~
 !!! error TS2527: The inferred type of 'f1' references an inaccessible 'this' type. A type annotation is necessary.
+        ~~
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
+!!! related TS9034 declarationFiles.ts:36:5: Add a return type to the method
             return { a: this };
         }
         f2(): this[] {
             return [this];
         }
         f3() {
         ~~
 !!! error TS2527: The inferred type of 'f3' references an inaccessible 'this' type. A type annotation is necessary.
+        ~~
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
+!!! related TS9034 declarationFiles.ts:42:5: Add a return type to the method
             return [{ a: this }];
         }
         f4(): () => this {
             return () => this;
